orchestrator/services/processes.py

```diff
@@ -526,6 +535,50 @@ def resume_process(
     return resume_func(process, user_inputs=user_inputs, user=user, broadcast_func=broadcast_func)
 
 
+def continue_awaiting_process(
+    process: ProcessTable,
+    *,
+    token: str,
+    input_data: State,
+) -> UUID:
+    """Continue a process awaiting data from a callback.
+
+    Args:
+        process: Process from database
+        token: The token which was generated for the process. This must match.
+        input_data: Data posted to the callback
+
+    Returns:
+        process id
+
+    """
+
+    pstat = load_process(process)
+    state = pstat.state.unwrap()
+
+    # Check if the token matches
+    token_from_state = state.get("__callback_token")
+    if token != token_from_state:
+        raise AssertionError("Invalid token")
+
+    # We need to pass the callback data to the worker executor. Currently, this is not supported.
+    # Therefore, we update the step state in the db and kick-off resume_workflow
+    # TODO: Allow passing additional data to be merged to the state upon resume_workflow
+    if "__callback_result_key" in state:
+        state = {**state, state["__callback_result_key"]: input_data}
+    else:
+        state = {**state, **input_data}
```

Comment:
What happens when we have a callback step without a result key, and the other system (accidentally) sends back an array instead of a dict?

Reply:
I think the callback result should always be stored in a key to avoid TypeErrors (or verify that the json data is an object/dict)

Reply:
:boom: 

In that case, what should be the preferred action? Maybe always add it under a default result key.

Reply:
Yeah, agree